src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,174 @@
+package Duke;
+
+import java.io.IOException;
+
+/**
+ * Duke.Duke is a basic chat-bot stores tasks into a list.
+ *
+ * @author Chi Song Yi Amadeus
+ * @version A-MoreOOP
+ * @since 17-08-2022
+ */
+public class Duke {
+
+    /**
+     * storage object deals with read/write interactions with file
+     */
+    private final Storage storage;
+
+    /**
+     * tasks manages a list of Task objects
+     */
+    private TaskList tasks;
+
+    /**
+     * ui handles user inputs and outputs
+     */
+    private final Ui ui;
+
+    /**
+     * parser interpretes user commands
+     */
+    private final Parser parser;
+
+    /**
+     * possible user commands.
+     */
+    enum Commands {
+        BYE,
+        LIST,
+        MARK,
+        UNMARK,
+        DELETE,
+        CREATETASK,
+        FIND
+    }
+
+    /**
+     * Constructor for Duke.
+     *
+     * @param filePath filepath to data file.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        parser = new Parser();
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            ui.showErrorReadingMessage();
+        }
+    }
+
+    /**
+     * Main method initializes welcome message, and then calls taskList method.
+     *
+     * @param args unused
+     */
+    public static void main(String[] args) {
+        new Duke("data/duke.txt").run();
+    }
+
+    /**
+     * TaskList method creates an input loop, creating Duke.Task objects and adding it to the array.
+     *
+     * @throws NumberFormatException if User inputs a non integer after mark/unmark
+     */
+    public void run() throws NumberFormatException {
+        while (ui.hasInput()) {
+            String input = ui.getUserInput().trim();
+            Commands command = parser.parseCommand(input);
+            if (command == Commands.BYE) {
+                break;
+            } else {
+                executeCommand(command, input);
+            }
+        }
+        ui.showGoodbyeMessage();
+    }
+
+
+    /**
+     * Processes and runs user commands.
+     *
+     * @param command user command
+     * @param input   input string
+     */
+    private void executeCommand(Commands command, String input) {
+        if (command == Commands.LIST) {
+            System.out.println("    ____________________________________________________________\n");
+            for (int i = 1; i <= tasks.getSize(); i++) {
+                String index = String.format("%d.", i);
+                String entry = index + tasks.getItem(i).toString();
+                ui.showEntry(entry);
+
+            }
+            System.out.println("    ____________________________________________________________\n");
+
+        } else if (command == Commands.MARK) {
```

Comment:
I think that there may be a lot of different cases going on here, so it may be a good idea to leave a comment on what case each block is handling!